### ISSUE TITLE
Unable to reach service backend running on remote node from host network

### ISSUE THREAD

**Issue by seanmwinn** [cilium/cilium#12244]
## Bug report

Newly installed cluster with 1.8.0 and managed etcd fails to start new servers due to failed TLS client auth settings.

**General Information**
On a newly installed cluster, etcd is failing to start due to a known bug in version 3.3 related to client TLS authentication: https://github.com/etcd-io/etcd/issues/9398

```
helm install cilium cilium/cilium \
   --version 1.8.0 \
   --namespace kube-system \
   --set global.nodeinit.enabled=true \
   --set global.kubeProxyReplacement=partial \
   --set global.hostServices.enabled=false \
   --set global.externalIPs.enabled=true \
   --set global.nodePort.enabled=true \
   --set global.hostPort.enabled=true \
   --set global.pullPolicy=IfNotPresent \
   --set config.ipam=kubernetes \
   --set global.hubble.enabled=true \
   --set global.hubble.listenAddress=":4244" \
   --set global.hubble.metrics.enabled="{dns,drop,tcp,flow,port-distribution,icmp,http}" \
   --set global.hubble.relay.enabled=true \
   --set global.hubble.ui.enabled=true \
   --set global.etcd.enabled=true \
   --set global.etcd.managed=true \
   --set global.etcd.clusterSize="1" \
   --set global.identityAllocationMode=kvstore \
   --set global.cluster.name=cluster-1 \
   --set global.cluster.id=1 \
   --set hubble-ui.image.tag=v0.6.1
```

Etcd server log contains:
```
WARNING: 2020/06/23 20:29:52 Failed to dial 0.0.0.0:2379: connection error: desc = "transport: authentication handshake failed: remote error: tls: bad certificate"; please retry.
```

[cilium-sysdump-20200623-133634.zip](https://github.com/cilium/cilium/files/4821797/cilium-sysdump-20200623-133634.zip)

**Comment by aanm** [comment]
@seanmwinn thanks for opening the issue, can you try with either `--set global.hostServices.enabled=true` or `--set global.kubeProxyReplacement=disabled`?

**Comment by aanm** [comment]
@borkmann @brb I was able to replicate this issue in 5.6 on dev VMs. When we run with `hostService.enabled=true` we can see the source IP to be the one from the cilium_host interface.
```
10.17.152.59.41980 > 10.16.226.130.2379
```

with `kube-proxy-replacement=disabled` the packets were still from `10.17.152.59.41980`.

I had to delete Cilium's DS and delete `cilium_host` interface to see the packet coming from `10.0.2.15` which is the same behavior that happens when running with the options:
```
   --set global.kubeProxyReplacement=partial \
   --set global.hostServices.enabled=false \
   --set global.externalIPs.enabled=true \
   --set global.nodePort.enabled=true \
   --set global.hostPort.enabled=true \
```

This makes the packet to be dropped because the packets are being set to an interface that can't reach the backends running on the other node. Is likely the routing logic happens before the service translation occurs.
